From d72ffd74f70cbea164e6d78e3dddcad7444696f2 Mon Sep 17 00:00:00 2001
From: Rabeeh Khoury <rabeeh@solid-run.com>
Date: Sun, 11 May 2014 22:25:05 +0300
Subject: [PATCH] Changes to Makefile to build SPL u-boot

SPL u-boot unifies CuBox-i and HummingBoard on all their versions

Signed-off-by: Rabeeh Khoury <rabeeh@solid-run.com>
---
 core/Makefile |   16 ++++++----------
 1 file changed, 6 insertions(+), 10 deletions(-)

diff --git a/core/Makefile b/core/Makefile
index 8aae659..76a583f 100644
--- a/core/Makefile
+++ b/core/Makefile
@@ -766,7 +766,7 @@ ifneq ($(strip $(TARGET_BOOTLOADER_CONFIG)),)
 BOOTLOADER_CROSS_TOOLCHAIN := `pwd`/prebuilts/gcc/linux-x86/arm/arm-eabi-4.6/bin/arm-eabi-
 BOOTLOADER_ENV := ARCH=arm CROSS_COMPILE=$(BOOTLOADER_CROSS_TOOLCHAIN)
 HOST_PROCESSOR := $(shell cat /proc/cpuinfo | grep processor | wc -l)
-TARGET_BOOTLOADER_IMAGE := $(PRODUCT_OUT)/u-boot.bin
+TARGET_BOOTLOADER_IMAGE := $(PRODUCT_OUT)/u-boot.img $(PRODUCT_OUT)/SPL
 
 BOOTLOADER_PATH := bootable/bootloader/uboot-imx/
 BOOTLOADER_CONFIG_FILE := $(BOOTLOADER_PATH)/include/config.h
@@ -774,15 +774,11 @@ BOOTLOADER_CONFIG_FILE := $(BOOTLOADER_PATH)/include/config.h
 bootloader: $(TARGET_BOOTLOADER_IMAGE)
 
 $(TARGET_BOOTLOADER_IMAGE):
-	@for ubootplat in $(TARGET_BOOTLOADER_CONFIG); do \
-		UBOOT_PLATFORM=`echo $$ubootplat | cut -d':' -f1`; \
-		UBOOT_CONFIG=`echo $$ubootplat | cut -d':' -f2`; \
-		$(MAKE) -C bootable/bootloader/uboot-imx/ distclean $(BOOTLOADER_ENV); \
-		$(MAKE) -C bootable/bootloader/uboot-imx/ $$UBOOT_CONFIG $(BOOTLOADER_ENV); \
-		$(MAKE) -C bootable/bootloader/uboot-imx/ $(BOOTLOADER_ENV); \
-		install -D bootable/bootloader/uboot-imx/u-boot.bin $(PRODUCT_OUT)/u-boot-$$UBOOT_PLATFORM.bin; \
-		install -D bootable/bootloader/uboot-imx/u-boot.bin $@; \
-	done
+	$(MAKE) -C bootable/bootloader/uboot-imx/ distclean $(BOOTLOADER_ENV);
+	$(MAKE) -C bootable/bootloader/uboot-imx/ $(TARGET_BOOTLOADER_CONFIG);
+	$(MAKE) -C bootable/bootloader/uboot-imx/ $(BOOTLOADER_ENV);
+	install -D bootable/bootloader/uboot-imx/u-boot.img $(PRODUCT_OUT)/u-boot.img
+	install -D bootable/bootloader/uboot-imx/SPL $(PRODUCT_OUT)/SPL
 else
 bootloader:
 endif
-- 
1.7.9.5

